src/client/manager.ts

```diff
@@ -18,61 +20,112 @@ export class ClientManager {
     }
 
-    /**
-     * Get all available tools from all connected clients
-     * @returns Promise resolving to a map of tool names to tool definitions
-     */
-    async getAllTools(): Promise<ToolSet> {
-        let allTools: ToolSet = {};
+    private clearClientCache(clientName: string): void {
+        const client = this.clients.get(clientName);
+        if (!client) return;
 
-        // Clear existing maps to avoid stale entries
-        this.toolToClientMap.clear();
+        [this.toolToClientMap, this.promptToClientMap, this.resourceToClientMap].forEach(cacheMap => {
+            for (const [key, mappedClient] of cacheMap.entries()) {
+                if (mappedClient === client) {
+                    cacheMap.delete(key);
+                }
+            }
+        });
+        logger.debug(`Cleared cache for client: ${clientName}`);
+    }
 
-        for (const [serverName, client] of this.clients.entries()) {
-            try {
-                logger.debug(`Getting tools from ${serverName}`);
-                const toolList = await client.getTools();
+    private async updateClientCache(clientName: string, client: IMCPClient): Promise<void> {
+        // Cache tools
+        try {
+            const tools = await client.getTools();
+            for (const toolName in tools) {
+                this.toolToClientMap.set(toolName, client);
+            }
+            logger.debug(`Cached tools for client: ${clientName}`);
+        } catch (error) {
+            logger.error(`Error retrieving tools for client ${clientName}:`, error);
+        }
 
-                // Map each tool to its provider client
-                for (const toolName in toolList) {
-                    this.toolToClientMap.set(toolName, client);
-                }
+        // Cache prompts, if supported
+        try {
+            const prompts = await client.listPrompts();
+            prompts.forEach(promptName => {
+                this.promptToClientMap.set(promptName, client);
+            });
+            logger.debug(`Cached prompts for client: ${clientName}`);
+        } catch (error) {
+            logger.debug(`Skipping prompts for client ${clientName}: ${error}`);
+        }
 
-                allTools = { ...allTools, ...toolList };
-                logger.debug(`Successfully got tools from ${serverName}`);
-            } catch (error) {
-                console.error(`Error getting tools from ${serverName}:`, error);
+        // Cache resources, if supported
+        try {
+            const resources = await client.listResources();
+            resources.forEach(resourceUri => {
+                this.resourceToClientMap.set(resourceUri, client);
+            });
+            logger.debug(`Cached resources for client: ${clientName}`);
+        } catch (error) {
+            logger.debug(`Skipping resources for client ${clientName}: ${error}`);
+        }
+    }
+
+    //Not used at the moment
+    private async rebuildAllCaches(): Promise<void> {
```

Comment:
seems like this isn't being used anywhere, we can delete it and add it back if/when we need it